src/main/com/fulcrologic/fulcro/algorithms/transit_old.cljc

```diff
@@ -0,0 +1,109 @@
+   and can be extended to support additional application-specific data types."
+  #?(:clj
+     (:refer-clojure :exclude [ref]))
+  (:require [cognitect.transit :as t]
+            #?(:cljs [com.cognitect.transit :as ct])
+            [com.fulcrologic.fulcro.algorithms.tempid :as tempid #?@(:cljs [:refer [TempId]])])
+  #?(:clj
+     (:import [com.cognitect.transit
+               TransitFactory WriteHandler ReadHandler]
+              [com.fulcrologic.fulcro.algorithms.tempid TempId])))
+
+#?(:cljs
+   (deftype TempIdHandler []
+            Object
+            (tag [_ _] tempid/tag)
+            (rep [_ r] (. r -id))
+            (stringRep [_ _] nil)))
+
+#?(:clj
+   (deftype TempIdHandler []
+     WriteHandler
+     (tag [_ _] tempid/tag)
+     (rep [_ r] (.-id ^TempId r))
+     (stringRep [_ r] (str tempid/tag "#" r))
+     (getVerboseHandler [_] nil)))
+
+#?(:cljs
+   (defn writer
+         "Create a transit writer.
+         - `out`: An acceptable output for transit writers.
+         - `opts`: (optional) options to pass to `cognitect.transit/writer` (such as handlers)."
+         ([]
+          (writer {}))
+         ([opts]
+          (t/writer :json
+                    (assoc-in opts [:handlers TempId] (TempIdHandler.))))))
+
+#?(:clj
+   (defn writer
+     "Create a transit writer.
+     - `out`: An acceptable output for transit writers.
+     - `opts`: (optional) options to pass to `cognitect.transit/writer` (such as data type handlers)."
+     ([out]
+      (writer out {}))
+     ([out opts]
+      (t/writer out :json
+                (assoc-in opts [:handlers TempId] (TempIdHandler.))))))
+
+#?(:cljs
+   (defn reader
+         "Create a transit reader.
+         - `opts`: (optional) options to pass to `cognitect.transit/reader` (such as data type handlers)."
+         ([]
+          (reader {}))
+         ([opts]
+          (t/reader :json
+                    (assoc-in opts
+                              [:handlers tempid/tag]
+                              (fn [id] (tempid/tempid id)))))))
+
+#?(:clj
+   (defn reader
+     "Create a transit reader.
+     - `opts`: (optional) options to pass to `cognitect.transit/reader` (such as data type handlers)."
+     ([in]
+      (reader in {}))
+     ([in opts]
+      (t/reader in :json
+                (assoc-in opts
+                          [:handlers tempid/tag]
+                          (reify
+                            ReadHandler
+                            (fromRep [_ id] (TempId. id))))))))
+
+(defn serializable?
+  "Checks to see that the value in question can be serialized by the default fulcro writer by actually attempting to
+  serialize it.  This is *not* an efficient check."
+  [v]
+  #?(:clj  (try
+             (.write (writer (java.io.ByteArrayOutputStream.)) v)
+             true
+             (catch Exception e false))
+     :cljs (try
+             (.write (writer) v)
+             true
+             (catch :default e false))))
+
+(defn transit-clj->str
```

Comment:
Do we need these?  If you're doing SSR you really HAVE to use metadata, so these just won't work for their intended purpose.